components/tikv_kv/src/rocksdb_engine.rs

```diff
@@ -256,6 +250,10 @@ impl Engine for RocksEngine {
         box_try!(self.sched.schedule(Task::Snapshot(cb)));
         Ok(())
     }
+
+    fn mut_coprocessor(&mut self) -> Option<&mut CoprocessorHost<BaseRocksEngine>> {
```

Comment:
It's weird, and seems only used for test. I think you should change the test infra that you can get a coprocessor host instead of pouching a hole in engine.

Reply:
En... you are quite right. Let me try.

Reply:
I move related logics into `RocksEngine`. `Engine` trait is not changed now. PTAL~

Reply:
I move related logics into `RocksEngine`. `Engine` trait is not changed now. PTAL~